### ISSUE TITLE
Consistently styling global vs local blocks

### ISSUE THREAD

**Issue by SaxonF** [WordPress/gutenberg#47348]
## What problem does this address?
![image](https://user-images.githubusercontent.com/1072756/214010222-718a15e9-1bc4-4918-8529-0153571bfa77.png)

Styling a button locally vs globally results in two very different experiences which adds to a users cognitive load. This also applies to how variations are listed and styled. 

This work slots under the broad goal of reducing the editors overall complexity.

## What is your proposed solution?
![image](https://user-images.githubusercontent.com/1072756/214028989-9d51ccf2-f7ac-4307-a6d7-e7b4fc9e4caa.png)

Use the same approach for both global and local contexts when styling a block. We would indicate whether a block was being styled globally. Variations are also listed the same way and in future this could include an "add variation" feature.
tldr inspector should look mostly the same when styling a block globally vs locally except for an added indicator.

This is partially a follow-up [this issue](https://github.com/WordPress/gutenberg/issues/47026#issuecomment-1383362020).

cc @WordPress/gutenberg-design

**Comment by jasmussen** [comment]
It was a bit hard for me to grok the initial proposal, as it included a couple of separate enhancement suggestions (like breadcrumbs, or edge to edge preview), but if I get it right, the last two panels are meant to represent GS > Blocks > Button (global context) and the editor inspector with a Button selected (local context), is that right?

If yes, then I think the main take-away that I agree with is that right now the two are diverging especially in terms of context. 

* Editor inspector has Color / Typography / Layout panels, but also other panels like Border and Advanced.
* Global styles has _sections_ for Color, Typography and Layout, but no more. So any block controls, like border radius, that doesn't fit in those three, usually gets dumped in the Layout section.

The fact that the two are so close but still diverge, does suggest we should unify here as a first step. I mention this in [the initial box shadow designs too](https://github.com/WordPress/gutenberg/issues/44651#issuecomment-1316740361). If need be we can do the ellipsis panel behavior too, but that seems like it can be explored separately.

**Comment by youknowriad** [comment]
Some time ago I've created this related discussion https://github.com/WordPress/gutenberg/discussions/37064 which is basically the same thing but form a technical level.

**Comment by carolinan** [comment]
Related: https://github.com/WordPress/gutenberg/issues/47146
You expressed it better, so I can close 47146. 
The main points from that issue is:
- We need to call these variations the same thing on both screens.
- I think both screens should use the wide display of the variation name that is used in global styles, since currently in the block editor, the names get truncated.

**Comment by SaxonF** [comment]
@jasmussen I've updated the issue to hopefully clarify. Will spin up breadcrumbs as a separate issue.

tldr inspector should look mostly the same when styling a block globally vs locally except for an added indicator.

**Comment by youknowriad** [comment]
Looking at the proposed designs, I think there's an inconsistency in the first mockup (the top level global styles). "Typography", "Colors"... should also be the same as the ones that show up for a given element/block.

**Comment by SaxonF** [comment]
@youknowriad not sure I follow, could you please expand? 

Pasting a related thought from a separate thread:

----

Quick thought. The current GS IA looks like:

- Typography
   - Elements
- Colours
   - Palette
   - Elements
- Layout
- Blocks

This has always felt a tiny bit strange to me because I often just want to focus on styling an element in its entirety as opposed to focusing on a single style first.

An alternative might be:

- Presets
   - Typography
   - Colours
   - Spacing
   - Layout
   - etc
- Styles
   - Elements
   - Blocks

Define your presets/foundation _then_ apply them to your site.

**Comment by youknowriad** [comment]
If you click on "Typography" in the top level screen of the global styles you'd see the same "screen" we have right now if you click "Typography" within "Button" block or "button" element. The only difference is that the top level updates the "body" element if you like. 

So what I was saying is that these top level styles (typography, colors, layout) should also be combined in the same view that is similar to the one you're proposing for the "button" global view.

**Comment by youknowriad** [comment]
Question about the proposed designs above. If you take a look a the "colors" panel in the paragraph block in both editor and global styles you notice this:

 - In the regular block inspector, it shows: "Text", "Background" and "Links"
 - In the global styles right now, it shows: "Text", "Background", "Links" and "Headings".

Should we support all the elements supported in global styles at the local block level as well?

Also, in the global styles panel we allow you to edit the palette which is technically a "setting" not a "style". How do we account for that in the proposed designs?

**Comment by jameskoster** [comment]
> Also, in the global styles panel we allow you to edit the palette which is technically a "setting" not a "style". How do we account for that in the proposed designs?

I've always found it a bit strange that palette editing is given so much prominence when styling a block. A contextual edit link in the color picker might work better.

<img width="260" alt="Screenshot 2023-02-14 at 10 51 42" src="https://user-images.githubusercontent.com/846565/218714685-ee36325d-2ef7-4fdd-aa41-f6ae558c4808.png">

**Comment by youknowriad** [comment]
Personally, I'd prefer to avoid it entirely there and separate "styles" and "settings" entirely if that makes sense.

**Comment by jameskoster** [comment]
I think there are still occasions where you might want to hop from styling a block to editing a palette, but it's not common and probably needs some design exploration. Removing for now would be an enhancement imo.

**Comment by SaxonF** [comment]
@youknowriad @jameskoster the video [here](https://github.com/WordPress/gutenberg/issues/47369) shows what editing block level palette might look like (at around 35 sec mark) and it is basically what @jameskoster proposed.

**Comment by youknowriad** [comment]
In some global styles contexts (root context, paragraph block...) we can tweak the colors of "headings" within that context. And to do so, we choose a sub context (all headings, h1, h2,...) and then we can define either background color, or text color for this sub context. How would you represent that in a dropdown menu when you click "headings" (see the button in the following screenshot). 

Note that, this needs to be a dropdown menu and not a "sub screen" like now, because we want to unify with how block inspectors render the buttons and a sub screen you navigate to can't work on the block inspector.

<img width="266" alt="Screenshot 2023-03-08 at 8 55 41 AM" src="https://user-images.githubusercontent.com/272444/223654293-71d6a000-7b58-4df2-9469-bf5675d4b50e.png">

**Comment by youknowriad** [comment]
@SaxonF In global styles, we have an "additional CSS" panel for blocks, how would that look like in the new unified global styles view?
